### PR TITLE
Fixed #27221 -- Document how to escape a percent symbol in ugettext

### PR DESCRIPTION
**Pull request by HenryDangPRG**
https://code.djangoproject.com/ticket/27221

### CODE REVIEW

File: docs/topics/i18n/translation.txt [hunk 1]
```diff
@@ -1792,6 +1792,30 @@ That's it. Your translations are ready for use.
...
+
+   1. Add a second percent sign
+   
+      .. code-block:: po
+      
+         #, python-format
+         msgid "10%%"
+         msgstr ""
+
+   2. Replace ``#, python-format`` with ``xgettext:no-python-format``
```
Comment (burhan) [inline docs/topics/i18n/translation.txt]: I think this should be "with #, xgettext:no-python-format"

**Comment by HenryDangPRG** [comment]
Yeah, that's a typo.

It's fixed now.

**Comment by timgraham** [comment]
Did you see [Claude's comment](https://code.djangoproject.com/ticket/27221#comment:6) on the ticket? Please uncheck "Patch needs improvement" on the ticket after updating the PR.

**Comment by claudep** [comment]
The explanation is still not completely accurate. This issue is only happening in certain cases, most often when a literal percent is followed by a space and a 's', a 'd' or a 'i'. So we shouldn't recommend to always apply the workaround, but only in those cases. I would have written something around these lines (feel free to adapt):

gettext misdetecting python-format in strings with percent characters

In certain cases, gettext is wrongly flagging strings with `python-format` (for example the string _("10% interest")). This will then generate an error while compiling message files (something like: 'msgstr' is not a valid Python format string, unlike 'msgid'.). There are two known workarounds in such cases: (followed by your examples).

File: docs/topics/i18n/translation.txt [hunk 2]
```diff
@@ -1792,6 +1792,26 @@ That's it. Your translations are ready for use.
...
+
+        output = _("10%% interest)
+
+    You can also disable python formatting so that all percent signs are 
+    treated as literals::
+
+        from django.utils.translation import ugettext as _
+     
+        #, xgettext:no-python-format
+        output = _("10%% interest)
```
Comment (claudep) [inline docs/topics/i18n/translation.txt]: In that case, there should be only one `%`.

**Comment by HenryDangPRG** [comment]
Ah, yeah, it's the string conversion types that makes the python-format trigger. I just linked to Python's documentation for the whole list of possible conversion types that could make ugettext flag the string, rather than listing out the most common ones.

File: docs/topics/i18n/translation.txt [hunk 3]
```diff
@@ -1792,6 +1792,43 @@ That's it. Your translations are ready for use.
...
+        from django.utils.translation import ugettext as _
+
+        output = _("10%% interest)
+
+    You can also disable python formatting so that all percent signs are 
+    treated as literals::
+
+        from django.utils.translation import ugettext as _
+     
+        #, xgettext:no-python-format
```
Comment (claudep) [inline docs/topics/i18n/translation.txt]: Could you check if the comma is necessary here?

**Comment by claudep** [comment]
Almost ready. @timgraham, a final review?

**Comment by HenryDangPRG** [comment]
@claudep The comma is extraneous, yes. I've removed it.

**Comment by timgraham** [comment]
Language looks good, I pushed a few cosmetic edits. One thing I wasn't sure about is if this is the best place in the i18n docs for this note. For ease of navigating the docs, I think we should be preferring sections rather than admonition boxes so I made that change.

Currently the nearby sections are:
* Localization: how to create language files 
** Message files
** Compiling message files
** **Troubleshooting: ugettext() incorrectly detects python-format in strings with percent signs**
** Creating message files from JavaScript source code

Does it look okay?

**Comment by claudep** [comment]
I think it's a good idea.

**Pull request merged**